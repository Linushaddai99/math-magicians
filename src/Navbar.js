import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="container">
    <h1>Maths Magicians</h1>
    <ul>
      <li>
        <Link to="/">Home</Link>
      </li>
      <li className="has-border">
        <Link to="/calculator">Calculator</Link>
      </li>
      <li>
        <Link to="/quote">Quote</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
