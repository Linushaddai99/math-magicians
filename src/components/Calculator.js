import React from 'react';
import Button from './button';
import calculate from './logic.js/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  render() {
    const { total, next, operation } = this.state;

    const handleClick = (event) => {
      const results = calculate({ total, next, operation }, event.target.innerHTML);

      this.setState({
        total: results.total,
        next: results.next,
        operation: results.operation,
      });
    };

    return (
      <div className="calculator-div">
        <div className="result">
          <span>{total}</span>
          <span>{operation}</span>
          <span>{next}</span>
        </div>
        <Button content="AC" handleClick={handleClick} />
        <Button content="+/-" handleClick={handleClick} />
        <Button content="%" handleClick={handleClick} />
        <Button content="รท" classes="orange" handleClick={handleClick} />
        <Button content="7" handleClick={handleClick} />
        <Button content="8" handleClick={handleClick} />
        <Button content="9" handleClick={handleClick} />
        <Button content="x" classes="orange" handleClick={handleClick} />
        <Button content="4" handleClick={handleClick} />
        <Button content="5" handleClick={handleClick} />
        <Button content="6" handleClick={handleClick} />
        <Button content="-" classes="orange" handleClick={handleClick} />
        <Button content="1" handleClick={handleClick} />
        <Button content="2" handleClick={handleClick} />
        <Button content="3" handleClick={handleClick} />
        <Button content="+" classes="orange" handleClick={handleClick} />
        <Button content="0" classes="zero" handleClick={handleClick} />
        <Button content="." handleClick={handleClick} />
        <Button content="=" classes="orange" handleClick={handleClick} />
      </div>
    );
  }
}

export default Calculator;
