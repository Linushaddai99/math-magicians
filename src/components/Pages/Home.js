import React from 'react';

const Home = () => (
  <div className="container" id="home-page">
    <h2>Welcome to our Page!</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
      Maiores quibusdam, tenetur illum provident unde
      mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci,
      delectus rem accusantium harum iste
      voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas
      vitae est ducimus nisi quaerat
      ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis.
      Deserunt dolore ullam
      tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita
      aliquam cumque
      quaerat possimus ad. Cumque quas nemo voluptas, fugit vitae recusandae
      numquam dolore architecto
      {' '}
    </p>

    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
      Maiores quibusdam, tenetur illum provident unde
      mollitia animi consequuntur aut eius quia. Incidunt, fugit et
      adipisci, delectus rem accusantium harum iste
      voluptatum exercitationem necessitatibus quam! Iusto nihil porro
      voluptas vitae est ducimus nisi quaerat
      ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi
      dolor officiis. Deserunt dolore ullam
      tempora consequuntur dolores minima praesentium hic quod tenetur at,
      officia expedita aliquam cumque.
    </p>
  </div>
);

export default Home;
