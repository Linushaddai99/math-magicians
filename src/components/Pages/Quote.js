import React from 'react';

const Quote = () => (
  <div>
    <p>
      Mathematics is not just about numbers, equations, components or algorithms:
      it is about understanding. - William Paul Thurston
    </p>
  </div>
);

export default Quote;
