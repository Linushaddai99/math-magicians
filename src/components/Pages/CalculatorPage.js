import React from 'react';
import Calculator from '../Calculator/Calculator';

const CalculatorPage = () => (
  <div className="container" id="calc-page">
    <h2>Lets Do some Maths</h2>
    <Calculator />
  </div>
);

export default CalculatorPage;
