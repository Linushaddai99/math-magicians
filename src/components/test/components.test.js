import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import CalculatorPage from '../Pages/CalculatorPage';
import Calculator from '../Calculator/Calculator';
import Home from '../Pages/Home';
import Quote from '../Pages/Quote';
import Navbar from '../../Navbar';

describe('testing components', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CalculatorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the home component correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the quote component correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Navbar component correctly', async () => {
    render(<Navbar />, { wrapper: BrowserRouter });
    expect(screen.queryByText(/Home/)).toMatchSnapshot();
  });

  it('stimulate users interaction', () => {
    render(<Calculator />);
    expect(screen.queryByText('AC')).toBeTruthy();
    fireEvent.click(screen.getByText(/AC/i));
  });
});
