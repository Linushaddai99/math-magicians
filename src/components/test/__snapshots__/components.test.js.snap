// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
  id="calc-page"
>
  <h2>
    Lets Do some Maths!
  </h2>
  <div
    className="calculator-div"
  >
    <div
      className="result"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="container"
  id="calc-page"
>
  <h2>
    Lets Do some Maths!
  </h2>
  <div
    className="calculator-div"
  >
    <div
      className="result"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`renders correctly 3`] = `
<div
  className="container"
  id="calc-page"
>
  <h2>
    Lets Do some Maths!
  </h2>
  <div
    className="calculator-div"
  >
    <div
      className="result"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`testing components renders  correctly 1`] = `
<div
  className="container"
  id="home-page"
>
  <h2>
    Welcome to our Page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque quaerat possimus ad. Cumque quas nemo voluptas, fugit vitae recusandae numquam dolore architecto
     
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque.
  </p>
</div>
`;

exports[`testing components renders correctly 1`] = `
<div
  className="container"
  id="calc-page"
>
  <h2>
    Lets Do some Maths!
  </h2>
  <div
    className="calculator-div"
  >
    <div
      className="result"
    >
      <span>
        0
      </span>
      <span />
      <span />
    </div>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="undefined"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`testing components renders correctly 2`] = `
<div
  className="container"
  id="home-page"
>
  <h2>
    Welcome to our Page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque quaerat possimus ad. Cumque quas nemo voluptas, fugit vitae recusandae numquam dolore architecto
     
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque.
  </p>
</div>
`;

exports[`testing components renders correctly 3`] = `
<div
  className="container"
  id="quote-page"
>
  <p>
    Mathematics is not just about numbers, equations, components or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;

exports[`testing components renders correctly 4`] = `
<a
  href="/"
>
  Home
</a>
`;

exports[`testing components renders the Navbar component correctly 1`] = `
<a
  href="/"
>
  Home
</a>
`;

exports[`testing components renders the home component correctly 1`] = `
<div
  className="container"
  id="home-page"
>
  <h2>
    Welcome to our Page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque quaerat possimus ad. Cumque quas nemo voluptas, fugit vitae recusandae numquam dolore architecto
     
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maiores quibusdam, tenetur illum provident unde mollitia animi consequuntur aut eius quia. Incidunt, fugit et adipisci, delectus rem accusantium harum iste voluptatum exercitationem necessitatibus quam! Iusto nihil porro voluptas vitae est ducimus nisi quaerat ipsa, culpa dolorem impedit aperiam id incidunt obcaecati eligendi dolor officiis. Deserunt dolore ullam tempora consequuntur dolores minima praesentium hic quod tenetur at, officia expedita aliquam cumque.
  </p>
</div>
`;

exports[`testing components renders the home component correctly 2`] = `
<div
  className="container"
  id="quote-page"
>
  <p>
    Mathematics is not just about numbers, equations, components or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;

exports[`testing components renders the quote component correctly 1`] = `
<div
  className="container"
  id="quote-page"
>
  <p>
    Mathematics is not just about numbers, equations, components or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;
